source/melpomene/src/sim_drivers/emb_display.rs

```diff
@@ -129,186 +136,170 @@ impl CommanderTask {
-                                .await
-                        }
-                        Err(e) => reply.reply_konly(req.reply_with(Err(e))).await,
+                }
+                Request::GetMeta => {
+                    let meta = DisplayMetadata {
+                        kind: FrameKind::Mono,
+                        width: self.width,
+                        height: self.height,
                     };
+                    let response = env.fill(Ok(Response::FrameMeta(meta)));
+                    let _ = reply_tx.reply_konly(response).await;
                 }
+                _ => todo!(),
             }
         }
     }
-}
 
-/// Create and return a Simulator display object from raw pixel data.
-///
-/// Pixel data is turned into a raw image, and then drawn onto a SimulatorDisplay object
-/// This is necessary as a e-g Window only accepts SimulatorDisplay object
-/// On a physical display, the raw pixel data can be sent over to the display directly
-/// Using the display's device interface
-fn frame_display(fc: &mut FrameChunk) -> Result<ImageRaw<Gray8>, ()> {
-    let raw_image: ImageRaw<Gray8>;
-    raw_image = ImageRaw::<Gray8>::new(fc.bytes.as_slice(), fc.width);
-    Ok(raw_image)
-}
+    /// Draw the given MonoChunk to the persistent framebuffer
+    async fn draw_mono(&self, fc: &MonoChunk, mutex: &Mutex<Option<Context>>) -> Result<(), ()> {
+        let mut guard = mutex.lock().await;
+        let ctx = if let Some(c) = (&mut *guard).as_mut() {
+            c
+        } else {
+            return Err(());
+        };
 
-struct FrameInfo {
-    frame: u16,
-}
+        let Context {
+            sdisp,
+            dirty,
+            framebuf,
+            ..
+        } = ctx;
 
-struct DisplayInfo {
-    frame_idx: u16,
-    frames: FixedVec<FrameInfo>,
+        draw_to(framebuf, fc, self.width, self.height);
+        let raw_img = frame_display(framebuf, self.width).unwrap();
+        let image = Image::new(&raw_img, Point::new(0, 0));
+        image.draw(sdisp).unwrap();
+        *dirty = true;
+
+        Ok(())
+    }
 }
 
-impl DisplayInfo {
-    // Returns a new frame chunk
-    async fn new_frame(
-        &mut self,
-        start_x: i32,
-        start_y: i32,
-        width: u32,
-        height: u32,
-    ) -> Result<FrameChunk, FrameError> {
-        let fidx = self.frame_idx;
-        self.frame_idx = self.frame_idx.wrapping_add(1);
+async fn render_loop(kernel: &'static Kernel, mutex: Arc<Mutex<Option<Context>>>) {
+    let mut idle_ticks = 0;
+    loop {
+        kernel.sleep(Duration::from_micros(1_000_000 / 15)).await;
+        let mut guard = mutex.lock().await;
+        let mut done = false;
+        if let Some(Context {
+            sdisp,
+            window,
+            dirty,
+            ..
+        }) = (&mut *guard).as_mut()
+        {
+            // If nothing has been drawn, only update the frame at 5Hz to save
```

Comment:
I was drawing at 1hz before, but it made MacOS sad and it would make the frame blink, likely greying it out when it thought the window was "unresponsive". I should look if there's an "update but don't redraw" option.